tests/probes/test_probes_topic.py

```diff
@@ -24,40 +24,45 @@
 ]
 
 
+@pytest.fixture
+def p_config():
+    return {"probes": {"topic": {"generations": 5}}}
+
+
 @pytest.mark.parametrize("probename", PROBES)
-def test_topic_wordnet_load(probename):
-    p = garak._plugins.load_plugin(probename)
+def test_topic_wordnet_load(probename, p_config):
```

Comment:
you know, it would be kinda cool if load_plugin() could work on probes without this. i can think of three reasons it might have been done the current way:

* unclear where that default should live
* unclear what that default should be
* resolution pending second PR

maybe there are more reasons, though. what do you think?

Reply:
If we just wanted this in tests, we could have the fixture set the value in `_config`.

I think the first two apply somewhat as currently the `default` location must be loaded at runtime by code in `cli` to promote it from `run` to the `probe`, maybe that responsibility should be moved.

A decision to set generations to `1` or `10` in the plugin class and warn that it was not provided via configuration passed vs the current exception could work, if `1` were used the `core` value or `10` being different from the fallback might be easy to spot if responsibility for the value gets unclear.

Happy to discuss this more.

Reply:
Did more thinking on this. We're converging (/have converged) on two possible ways to load a plugin:
1. Via CLI during a CLI-invoked run
2. Via `_plugins.load_plugin()`

Further, outside of testing, probes are only loaded by a harness - there aren't cases where they're instantiated by e.g. other probes. This is different from detectors or generators, which are occasionally instantiated outside of a harness during a garak run. This constrained set of contexts for probe instantiation makes our lives easier.

In case (1), CLI runs, `_config.run` should be populated and authoritative and the CLI will populate individual `probe.generations` values. So we don't need to worry about a default. Would be interesting to test if probes are ever invoked without one, in fact.

In case (2), we do need a default. The prior value, 10, was arbitrary, when we didn't know much about testing LLMs. Why not be conservative and fast and make the new default something smaller, which will play nice across the board - `1`? This would be set in the probe plugin class. Oh, maybe as a `DEFAULT_PARAM` default? Not sure of the implications of that.

We could try:
* If a probe is loaded using a config, expect a `generations` parameter, and raise ValueError if absent
* If a probe is loaded without a config, set `generations=1`

I suggest requiring that any config supplies `generations` in order to reduce instances of magically populating that number behind the scenes. I can also be talked out of this.

Reply:
I think default of `1` and exposing as a `DEFAULT_PARAM` would be a reasonable middle ground.  Adding a `detection` to require the value was passed in `config_root` would likely end up hard to keep consistent. An exposed default of `1` gives the consumer documentation of what will occur.

Reply:
Nice. Simple. Let's do that.